tests/api_fastapi/core_api/routes/public/test_variables.py

```diff
@@ -136,6 +136,34 @@ def test_get_should_respond_404(self, test_client):
         assert f"The Variable with key: `{TEST_VARIABLE_KEY}` was not found" == body["detail"]
 
 
+class TestGetVariables(TestVariableEndpoint):
+    @pytest.mark.enable_redact
+    @pytest.mark.parametrize(
+        "query_params, expected_total_entries, expected_keys",
+        [
+            # Filters
+            ({}, 3, [TEST_VARIABLE_KEY, TEST_VARIABLE_KEY2, TEST_VARIABLE_KEY3]),
+            ({"limit": 1}, 3, [TEST_VARIABLE_KEY]),
+            ({"limit": 1, "offset": 1}, 3, [TEST_VARIABLE_KEY2]),
+            # Sort
+            ({"order_by": "id"}, 3, [TEST_VARIABLE_KEY, TEST_VARIABLE_KEY2, TEST_VARIABLE_KEY3]),
+            ({"order_by": "-id"}, 3, [TEST_VARIABLE_KEY3, TEST_VARIABLE_KEY2, TEST_VARIABLE_KEY]),
+            ({"order_by": "key"}, 3, [TEST_VARIABLE_KEY3, TEST_VARIABLE_KEY2, TEST_VARIABLE_KEY]),
+            ({"order_by": "-key"}, 3, [TEST_VARIABLE_KEY, TEST_VARIABLE_KEY2, TEST_VARIABLE_KEY3]),
```

Comment:
I scratched my head here for a while. I didn't test `-value` and `value` because it appears that the ordering is operating on the encoded value in the db (with the fernet key), and not with the original value.

This does not appear to be the case when I run the server locally and test manually, just in tests, which I find really weird.

If it really order on the encoded fernet key value then this sort doesn't accomplish much. And on top of that the FERNET_KEY is regenerated on each test run, therefore the ordering is not stable.

If someone has an idea.

Reply:
>This does not appear to be the case when I run the server locally and test manually, just in tests, which I find really weird.

This could probably be because on local config it'd using the fernet key with `is_encrypted = False`, while in test config as you mentioned it's using a dynamically generated fernet key.

>If someone has an idea.

Since encrypted values are getting sorted, doesn't look like sort by `value` would be useful to the user. Also if values are large then sorting them could degrade performance.

Reply:
> This could probably be because on local config it'd using the fernet key with is_encrypted = False, while in test config as you mentioned it's using a dynamically generated fernet key.

Oh well that explains a lot. I just verified in the db and indeed values are not encrypted. For encrypted values (most business use case I would say), this sort is just irrelevant, I think I'll just remove it all together.